index2.js

```diff
@@ -0,0 +1,269 @@
+      let index = player.playerInventory.indexOf(lookUpItems[item]);
+      roomLookupTable[this.currentState].roomInventory.push(player.playerInventory[index]); 
+      player.playerInventory.splice(index, 1);;
+      console.log(`You dropped the ${item}`);
+      getInput();
+    } else {
+    console.log(`I'm sorry you can't drop the ${itemName}`);
+    getInput();
+   }
+  }
+
+  open(item){
+    if(lookUpItems[item].openText){
+    console.log(lookUpItems[item].openText);
+    }else{
+      console.log(`I'm sorry I can't open ${item}`)
+    }
+    getInput();
+  }
+
+  code(number){
+    if(roomLookupTable[this.currentState] === mainSt && number === '12345'){
+      console.log(`Success! The door opens. You enter the foyer and the door
+      shuts behind you.`);
+      enterState('foyer', 'mainSt');
+    }else if(roomLookupTable[this.currentState] === mainSt && number !== '12345'){
+      console.log(`Bzzzzt! The door is still locked.`);
+      getInput();
+    }else{
+      console.log('There is no keypad on this side.');
+      getInput();
+    }
+  }
+
+  inventory(){
+    if(player.playerInventory.length < 1){
+      console.log("You have nothing.")
+      return getInput();
+    } else {
+      player.playerInventory.forEach(function(element) {
+        console.log(element.inventoryText);
+      });
+      return getInput();
+    }
+  }
+  move(newState){
+    if(Object.keys(roomLookupTable).includes(newState) && this.currentState !== "mainSt"){
+    console.log(this.currentState);
+      enterState(newState, this.currentState);
+    console.log(this.currentState.description)
+    } else {
+      console.log(`I can't move to ${newState}`)
+    }
+   getInput();
+  }
+
+  async xyzzy(){
+    let newRoom = await ask("Which room would you like to go to mainSt or foyer?")
+    if(Object.keys(roomLookupTable).includes(newRoom)){
+      this.currentState = newRoom;
+      console.log(roomLookupTable[player.currentState].name);
+      console.log(roomLookupTable[player.currentState].description);
+    }else{
+          console.log(`That isn't a choice. Please try again`);
+          return xyzzy();
+  }
+   return getInput();
+  }
+}
+
+//Items
+const paper = new Item(
+"Seven Days", 
+"Some kind of newspaper with bands you've never heard of.", 
+true,
+`You pick up the paper and leaf through it looking for comics and ignoring the articles, just like everybody else does.`,
+"You are carrying Seven Days, Vermonts Alt-Weekly",
+"You flip through the pages. Nothing catches your eye."
+);
+
+const sign = new Item(
+  "Sign", 
+  `Welcome to Burlington Code Academy! Come on 
+up to the third floor. If the door is locked, use the code 
+12345`,
+  false,
+  "That would be selfish. How will other students find their way?",
+  "",
+  "",
+);
+
+const door = new Item(
+  "door",
+  "182 Main St",
+  false,
+  "It's locked and I don't know how you'd carry it.",
+  "",
+  "The door is locked. There is a keypad on the door handle."
+);
+
+//Rooms
```

Comment:
Great use of comments to divide your objects and classes into specific categories